pkgs/by-name/dp/dprint/plugins/default.nix

```
@@ -45,5 +53,11 @@ let
     name: _:
     nameValuePair (removeSuffix ".nix" name) (import (./. + "/${name}") { inherit mkDprintPlugin; })
   ) files;
+  # cb arg: can pass a callback to filter out plugins from nixpkgs or a list of plugins
+  getPluginList =
+    cb:
+    map (p: "${builtins.toString p}/plugin.wasm") (
+      builtins.filter isDerivation (if builtins.isFunction cb then cb plugins else cb)
+    );
```

Not a strong preference but, usually I would avoid polymorphic function arguments. From a userperspective this is not much harder to use and documentation gets actually shorter.

```suggestion
  getPluginList =
    cb:
    map (p: "${builtins.toString p}/plugin.wasm") (
      builtins.filter isDerivation (cb plugins)
    );
```

---

Why do we actually filter out derivations btw? It feels a bit like discarding errors?

---

Or this:

```suggestion
  getPluginList =
    cb:
    let
      plugins' = cb plugins;
    in
    map (p: "${builtins.toString p}/plugin.wasm") (
      builtins.filter isDerivation plugins'
    ) ++ builtins.filter (!isDerivation) plugins';
```

But I think I actually prefer the simpler version.

---

> With this change if I were to maintain a dprint plugin out of tree, outside nixpkgs, I need to reimplement `getPluginList` if I wish to use `mkDprintPlugin`

Can you explain why this is the case? You are not forced to use values from `ps`, if I am not mistaken.

---

so I can use this

```nix
pkgs.dprint-plugins.getPluginList (builtins.attrValues pkgs.dprint-plugins)
```

while filtering out `getPluginList, mkDprintPlugin, override, overrideDerivation, recurseForDerivations`.

But I don' think it is needed anymore. will remove.